src/braket/tasks/gate_model_quantum_task_result.py

```diff
@@ -151,6 +151,36 @@ def get_compiled_circuit(self) -> Optional[str]:
         else:
             return None
 
+    def partial_measurements(self, qubits: Iterable[int]) -> np.ndarray:
+        """
+        Gets the partial measurements from the qubits given
+
+        Args:
+            qubits (Iterable [int]): an array of qubits
+
+        Returns:
+            ndarray: an array of the measurements of the given qubits
+        """
```

Comment:
Let's make sure to clarify that this is a post-processing filter, rather than a hardware partial measurement. Maybe something like "Gets a filtered list of partial measurements over the qubits provided."

and nit: I would opt for "list" over "array" in a Python context

Reply:
Sorry, I'm confused. 
* Why are we introducing a post-processing filter? Don't we already have a way to do this using the `sample` result type? 
* I'd have expected a `measure` instruction to be introduced instead. What prevents us from doing that?

Reply:
Correct. That is still the plan, but will require service support. This is meant to be a convenience method for the result type class. We can rename/hold off if you think it will cause confusion.

Sample result type does give us the same information, but the results are given as [-1, +1] along a given observable, rather than the the familiar 0/1.

Reply:
I believe service support is already present. You should be able to run a circuit which measures a subset of the qubits:
```
oq = """
OPENQASM 3.0;
bit[1] b;
qubit[2] q;
h q[0];
cnot q[0], q[1];
b[0] = measure q[0];
"""

from braket.ir.openqasm import Program as OpenQASMProgram
p = OpenQASMProgram(source=oq)

oqc = AwsDevice("arn:aws:braket:eu-west-2::device/qpu/oqc/Lucy")
oqc.run(p)
```


The device property `requiresAllQubitsMeasurement` specifies whether you can measure a subset of the qubits or not.